TFNetworkRecLayer.py

```diff
@@ -6116,3 +6116,60 @@ def __call__(self, inputs, state, scope=None):
     new_state = tf.nn.rnn_cell.LSTMStateTuple(c, h)
 
     return output, new_state
+
+class RelativePositionalEncodingLayer(_ConcatInputLayer):
+  """
+  Relative positioning term as introduced by Shaw et al., 2018
+
+  Usually added to Self-Attention using key_shift.
+  Parts of the code are adapted from Tensor2Tensor (https://github.com/tensorflow/tensor2tensor).
+  """
+  layer_class = "relative_positional_encoding"
+  recurrent = True
+
+  def __init__(self, n_out, forward_weights_init="glorot_uniform", clipping=16, fixed=False, **kwargs):
+    """
+    :param n_out: Feature dimension of encoding.
+    :param clipping: After which distance to fallback to the last encoding
+    :param fixed: Uses sinusoid positional encoding instead of learned parameters
```

Comment:
The argument types are not specified (`int n_out`, `int clipping`, `bool fixed`). Always specify them if possible.
Also, you don't specify all arguments (`forward_weights_init` is missing).